### PR TITLE
[BB-3262] Add cleanup tmp task and tag

### PR DESCRIPTION
**Pull request by AdditionalPylons**
For BB-3262: Added ansible task to cleanup old mattermost installer files from /tmp after updating.

**Dependencies**: None

**Merge deadline**: None

**Testing instructions**:

1. Make copies of mattermost installer files in /tmp and change file names or redownload. The ansible task saves the 2 most recent versions of the autolink and teams installers, so there must be 3+ of each in the folder to test cleanup.
2. Run playbook with "--tags cleanup_tmp" and check that only the 2 most recent versions of each installer remains in /tmp. Check that disk space is at an acceptable level (i.e. around 50% usage).
3. Run playbook again without any extra installer files in /tmp to cleanup. Make sure that no errors are thrown.

**Reviewers**
- [ ] @0x29a

### CODE REVIEW

File: playbooks/roles/mattermost/tasks/mattermost.yml
Comment (toxinu) [inline playbooks/roles/mattermost/tasks/mattermost.yml:141]: @AdditionalPylons Can you try to use the built-in Ansible tools like [find](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/find_module.html) along with [file](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html)? It will allow the task to be more resilient to changes.

Comment (AdditionalPylons) [reply playbooks/roles/mattermost/tasks/mattermost.yml:141]: @toxinu Please see https://tasks.opencraft.com/browse/BB-3262?focusedCommentId=179148&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-179148 for why I used shell instead of find. Maybe there is a better way to handle it that I'm not aware of? I'm pretty new to writing ansible tasks

Comment (AdditionalPylons) [reply playbooks/roles/mattermost/tasks/mattermost.yml]: @0x29a and @toxinu Please take a look at the updated PR. Since this is my first ansible story, I'm not really sure of a good way to test it locally, but I believe this ought to do the trick.

**Review by toxinu — APPROVED** [review]
Nice work! Looks good to me 👍

**Comment by AdditionalPylons** [comment]
Awesome! I haven't been able to test it, so I'll wait for @0x29a to sign off after testing before merging.

**Review by 0x29a — CHANGES_REQUESTED** [review]
@bill, I tested these changes, and they require some improvements.

First, tags for new tasks seems to be useless since they're nested in block, so when I try to run only them I get this:
```
TASK [mattermost : find old Mattermost installer files in /tmp] ***************************************************************************************************************************************************
task path: /home/acid/Projects/ansible-secrets/deploy/playbooks/roles/mattermost/tasks/mattermost.yml:140
fatal: [chat.opencraft.com]: FAILED! => {
    "msg": "The conditional check 'mattermost_version.changed or not mattermost_installation_complete.stat.exists' failed. The error was: error while evaluating conditional (mattermost_version.changed or not mattermost_installation_complete.stat.exists): 'mattermost_version' is undefined\n\nThe error appears to have been in '/home/acid/Projects/ansible-secrets/deploy/playbooks/roles/mattermost/tasks/mattermost.yml': line 140, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: find old Mattermost installer files in /tmp\n      ^ here\n"
}
	to retry, use: --limit @/home/acid/Projects/ansible-secrets/deploy/playbooks/chat.retry
```

Second, our Ansible version doesn't support some of the `find` module args you used:
```
TASK [mattermost : find old Mattermost installer files in /tmp] ***************************************************************************************************************************************************
task path: /home/acid/Projects/ansible-secrets/deploy/playbooks/roles/mattermost/tasks/mattermost.yml:140
Using module file /home/acid/.pyenv/versions/3.6.11/envs/ansible/lib/python3.6/site-packages/ansible/modules/files/find.py
<chat.opencraft.com> ESTABLISH SSH CONNECTION FOR USER: ubuntu
<chat.opencraft.com> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/home/acid/.ssh/dedicated.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ubuntu -o ConnectTimeout=10 -o ControlPath=/home/acid/.ansible/cp/f91a02391e chat.opencraft.com '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-kcifplfmycsisqfvjnnluchfpynbeuuv; /usr/bin/python3'"'"'"'"'"'"'"'"' && sleep 0'"'"''
Escalation succeeded
<chat.opencraft.com> (1, b'\n{"failed": true, "invocation": {"module_args": {"paths": "/tmp", "recurse": false, "register": "all_mm_team", "patterns": "mattermost-team*.tar.gz"}}, "msg": "Unsupported parameters for (find) module: register Supported parameters include: age, age_stamp, contains, excludes, file_type, follow, get_checksum, hidden, paths, patterns, recurse, size, use_regex"}\n', b'')
fatal: [chat.opencraft.com]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "paths": "/tmp",
            "patterns": "mattermost-team*.tar.gz",
            "recurse": false,
            "register": "all_mm_team"
        }
    },
    "msg": "Unsupported parameters for (find) module: register Supported parameters include: age, age_stamp, contains, excludes, file_type, follow, get_checksum, hidden, paths, patterns, recurse, size, use_regex"
}
	to retry, use: --limit @/home/acid/Projects/ansible-secrets/deploy/playbooks/chat.retry

PLAY RECAP ********************************************************************************************************************************************************************************************************
chat.opencraft.com         : ok=1    changed=0    unreachable=0    failed=1   
```

Here is the version we use:
```
(ansible) ➜  ansible-secrets git:(master) ✗ ansible-playbook --version
ansible-playbook 2.5.1
  config file = /home/acid/Projects/ansible-secrets/ansible.cfg
  configured module search path = ['/home/acid/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/acid/.pyenv/versions/3.6.11/envs/ansible/lib/python3.6/site-packages/ansible
  executable location = /home/acid/.pyenv/versions/ansible/bin/ansible-playbook
  python version = 3.6.11 (default, Aug 19 2020, 12:44:52) [GCC 9.3.0]
```

**Comment by AdditionalPylons** [comment]
Thanks for helping to test this @0x29a, I'll rework and resubmit this later today.

PS you tagged the wrong Bill :P

**Comment by 0x29a** [comment]
Ah, @Bill, sorry for disturbance :slightly_smiling_face: 

@AdditionalPylons, okay, sounds good.

**Comment by AdditionalPylons** [comment]
@0x29a There was an indent problem with register in the last commit that I've since fixed and I gave the cleanup code its own block, so running the tag to test should work now. Please re-try.

**Comment by 0x29a** [comment]
@AdditionalPylons, I tested recent changes, here is what I got:
```
TASK [mattermost : select all but the latest 2 installers for cleanup] ********************************************************************************************************************************************
task path: /home/acid/Projects/ansible-secrets/deploy/playbooks/roles/mattermost/tasks/mattermost.yml:156
fatal: [chat.opencraft.com]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: 'all_mm_autolink' is undefined\n\nThe error appears to have been in '/home/acid/Projects/ansible-secrets/deploy/playbooks/roles/mattermost/tasks/mattermost.yml': line 156, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: select all but the latest 2 installers for cleanup\n      ^ here\n"
}
	to retry, use: --limit @/home/acid/Projects/ansible-secrets/deploy/playbooks/chat.retry
```

Since this task doesn't depend on other playbook content, could you test it against Vagrant box to avoid review round-trips?
Just use `ansible==2.5.1` and Ubuntu 16.04. Sorry for not suggesting this earlier.

**Comment by AdditionalPylons** [comment]
Sorry for taking so long @0x29a, debugging this was a lot harder than expected. Finally got everything worked out though, it's been tested on a vagrant box locally and correctly deletes all but the 2 newest installers of each type, doesn't affect any other files or folders, and doesn't throw any errors if the /tmp folder is empty or has less than 2 installers. Please run the test yourself but this one should finally be good to merge.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I ran these tasks against our chat server and it deleted old installers.
- [x] I read through the code
- [ ] Includes documentation

**Pull request merged**
